File: Cono.m
bandera=1;
while bandera==1;
try
a=input('Ingrese radio del cono:');
b=input('Ingrese generatriz del cono:');
c=input('Ingrese altura del cono:');
entrada=1;
catch
printf('Error ingrese un texto del menú \n');
entrada=0;
end_try_catch 

  if(entrada=1)
    base=pi*a^2;
    arealateral=pi*a*b;
    areatotal=base+arealateral;
    volumen=b*c*a*a/3;
 
    archivo=fopen('Registro_cono.txt','a');
    fprintf(archivo,'Area de la base=: %i');
    fprintf(archivo,' %i',base); 
    fprintf(archivo,'Area lateral=: %i');
    fprintf(archivo,' %i',arealateral);   
    fprintf(archivo,'Area Total=: %i');
    fprintf(archivo,' %i',areatotal);  
    fprintf(archivo,' \n');  
    fclose(archivo);
    printf("El area de la base es:%d",base);
    printf("\n");
    printf("El area lateral es:%d",arealateral);
    printf("\n");
    printf("El area Total es:%d",areatotal);
    printf("\n");
    printf("El volumen del cono es: %d",volumen);
    printf('\n');
  else
    printf("Error, ingrese valores numericos\n");
  endif 
  
  
back = input('Desea Continuar s/n:','s');
            if (back=='s')
                bandera=1;
            elseif(back=='n')
                bandera=0;
            endif
end

File: cono.py
archivo=open("Cono.txt", "a")
import math

def cono():
    print("Calculos para el Cono.\n")
    print("1.Area base.\n2.Area lateral.\n3.Area total.\n4.Volumen.\n")
    x=int(input("Escoja una operacion: "))


    if x==1:
        r=float(input("Ingrese el radio del cono: "))
        g=float(input("Ingrese la generatriz del cono: "))
        h=float(input("Ingrese la altura del cono: "))

        abase=math.pi*r**2
        print("El área de la base es: ", abase)
        archivo.write('abase=% s' %abase + '\n')

    if x==2:
        r=float(input("Ingrese el radio del cono: "))
        g=float(input("Ingrese la generatriz del cono: "))
        h=float(input("Ingrese la altura del cono: "))
            
        alateral=math.pi*r*g
        print("El area lateral es: ", alateral)
        archivo.write('alateral=% s' %alateral + '\n')

    if x==3:
        r=float(input("Ingrese el radio del cono: "))
        g=float(input("Ingrese la generatriz del cono: "))
        h=float(input("Ingrese la altura del cono: "))

        atotal=abase+alateral
        print("El área Total es: ", atotal)
        archivo.write('atotal=% s' %atotal + '\n')

    if x==4:
        r=float(input("Ingrese el radio del cono: "))
        g=float(input("Ingrese la generatriz del cono: "))
        h=float(input("Ingrese la altura del cono: "))

        volumen=math.pi*r**2*h/3

        print("El volumen del cono es: ", volumen)
        archivo.write('volumen=% s' %volumen + '\n')

##continuar
continua=True
while (continua):
    seleccion=cono()
    print("¿Quieres continuar (s/n)? ")
    if (input ()=="s" or input ()=="S"):
        continua=True
    else:
        continua=False
        print("Fin del programa")

archivo.close()
